nutridine/src/screens/macro/MacroPage.js

```diff
@@ -0,0 +1,214 @@
+import React, { useState, useEffect } from "react";
+import {
+  Button,
+  RangeSlider,
+  RangeSliderTrack,
+  RangeSliderFilledTrack,
+  RangeSliderThumb,
+  VStack,
+  Text,
+  HStack,
+  Input,
+  useColorModeValue,
+  Card,
+  CardBody,
+  Box,
+} from "@chakra-ui/react";
+import useMealsByMacros from "../../hooks/useMealsByMacros";
+import { useMeals } from "../../contexts/MealsContext";
+
+function AttributeSliders() {
+  const { setMeals } = useMeals();
+
+  const [calories, setCalories] = useState([200, 500]);
+  const [protein, setProtein] = useState([10, 50]);
+  const [carbs, setCarbs] = useState([20, 100]);
+  const [fat, setFat] = useState([5, 35]);
+
+  const [showSliders, setShowSliders] = useState(true);
+  const [shouldFetch, setShouldFetch] = useState(false);
+  const [desiredMacros, setDesiredMacros] = useState({
+    query: "fast food",
+    minCalories: 0,
+    maxCalories: 0,
+    minProtein: 0,
+    maxProtein: 0,
+    minCarbs: 0,
+    maxCarbs: 0,
+    minFat: 0,
+    maxFat: 0,
+  });
+
+  const { meals, loading, error } = useMealsByMacros(
+    desiredMacros,
+    shouldFetch
+  );
+
+  const handleInputChange = (index, value, setValue, inputValue) => {
+    const newValue = [...value];
+    newValue[index] = Number(inputValue);
+    setValue(newValue);
+  };
+
+  const handleSubmit = () => {
+    setDesiredMacros({
+      query: "fast food",
+      minCalories: calories[0],
+      maxCalories: calories[1],
+      minProtein: protein[0],
+      maxProtein: protein[1],
+      minCarbs: carbs[0],
+      maxCarbs: carbs[1],
+      minFat: fat[0],
+      maxFat: fat[1],
+    });
+    setShowSliders(false);
+    setShouldFetch(true);
+  };
+
+  useEffect(() => {
+    if (!loading && meals) {
+      setMeals(meals);
+      console.log(meals);
```

Comment:
left in console log by accident

Reply:
This was intentional for salam to see them but im rendering them anyways, so im going to get rid of this